test/In.ProjectEKA.HipServiceTest/OpenMrs/DiscoveryDataSourceTest.cs

```diff
@@ -0,0 +1,253 @@
+
+            openmrsClientMock.Setup(x => x.GetAsync("ws/fhir2/Patient")).ReturnsAsync(new HttpResponseMessage
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = new StringContent(patientSample)
+                })
+                .Verifiable();
+
+            //When
+            var patients = await discoveryDataSource.LoadPatientsAsync(null, null, null);
+
+            //Then
+            var firstPatient = patients[0];
+            firstPatient.Name[0].GivenElement.First().ToString().Should().Be("Test");
+            firstPatient.Gender.Should().Be(AdministrativeGender.Female);
+            firstPatient.BirthDate.Should().Be("1982-05-05");
+            var secondPatient = patients[1];
+            secondPatient.Name[0].GivenElement.First().ToString().Should().Be("David");
+            secondPatient.Gender.Should().Be(AdministrativeGender.Male);
+            secondPatient.BirthDate.Should().Be("1997-04-10");
+        }
+
+        [Fact]
+        public async System.Threading.Tasks.Task ShouldReturnEmptyPatientWhenGotNoRecord()
+        {
+            //Given
+            var openmrsClientMock = new Mock<IOpenMrsClient>();
+            var discoveryDataSource = new DiscoveryDataSource(openmrsClientMock.Object);
+
+            openmrsClientMock.Setup(x => x.GetAsync("ws/fhir2/Patient")).ReturnsAsync(new HttpResponseMessage
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = new StringContent(notFoundData)
+                })
+                .Verifiable();
+
+            //When
+            var patients = await discoveryDataSource.LoadPatientsAsync(null, null, null);
+
+            //Then
+            patients.Count.Should().Be(0);
+        }
+
+        [Theory]
+        [InlineData("ws/fhir2/Patient/?name=David", "David", null, null)]
+        [InlineData("ws/fhir2/Patient/?gender=M", null, Gender.M, null)]
+        [InlineData("ws/fhir2/Patient/?gender=F", null, Gender.F, null)]
+        [InlineData("ws/fhir2/Patient/?birthdate=1982-05-05", null, null, "1982-05-05")]
+        public async System.Threading.Tasks.Task ShouldQueryDataSourceByNameAccordingToTheFilter(
+            string expectedPath, string name, Gender? gender, string yearOfBrith)
+        {
+            //Given
+            var openmrsClientMock = new Mock<IOpenMrsClient>();
+            var discoveryDataSource = new DiscoveryDataSource(openmrsClientMock.Object);
+
+            openmrsClientMock
+                .Setup(x => x.GetAsync(It.IsAny<string>()))
+                .ReturnsAsync(new HttpResponseMessage
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = new StringContent(patientSample)
+                })
+                .Verifiable();
+
+            //When
+            var patients = await discoveryDataSource.LoadPatientsAsync(name, gender, yearOfBrith);
```

Comment:
Codacy found an issue: [Remove the unused local variable 'patients'.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5859908)